### PR TITLE
GODRIVER-2277 Remove causal consistency prose test #10

### PR DESCRIPTION
**Pull request by prestonvasquez**
[GODRIVER-2277](https://jira.mongodb.org/browse/GODRIVER-2277)

Prose Test no. 10 in the [Causal Consistency Specification](https://github.com/mongodb/specifications/blob/master/source/causal-consistency/causal-consistency.rst#test-plan), to validate `nil` operation time when executed in a client session, is outdated per [SPEC-1019](https://github.com/mongodb/specifications/commit/517ec1da43db9458aaef389c60a943c1c9ae446e) which "prohibit[s] unack'd writes in a session".  

The work to implement the change in SPEC-1019 for the GODRIVER team was completed by the work done on [GODRIVER-52](https://jira.mongodb.org/browse/GODRIVER-52), specifically with [PR#77](https://github.com/mongodb/mongo-go-driver/pull/77/files#diff-6bd094ee6d491c2cdc5f5c001c899b47fe0388373a3833f191d5f26185d14351R129).  

Add a `TestSessions` subtest to ensure that mongo-go-driver errors on unacknowledged writes (i.e. a `0`-number write concern) during a session.

### Question(s)
- [x] I've verified through our prose test no. 10 that we get an error when trying to submit an unacknowledged write through a session.  Instead of removing the test entirely, do we want to tweak the test to assert that doing an unacknowledged write within a session results in an error?

### CODE REVIEW

File: mongo/integration/causal_consistency_test.go [hunk 1]
```diff
@@ -193,21 +192,6 @@ func TestCausalConsistency_Supported(t *testing.T) {
 		assert.NotNil(mt, sentOptime, "expected operation time on command, got nil")
 		assert.True(mt, currOptime.Equal(*sentOptime), "expected operation time %v, got %v", currOptime, sentOptime)
 	})
-	unackWcOpts := options.Collection().SetWriteConcern(writeconcern.New(writeconcern.W(0)))
-	mt.RunOpts("unacknowledged write", mtest.NewOptions().CollectionOptions(unackWcOpts), func(mt *mtest.T) {
```
Comment (benjirewis) [inline mongo/integration/causal_consistency_test.go:197]: Just curious, in what situations does this test fail? It does not fail locally for me against a 5.0 replica set.
> do we want to tweak the test to assert that doing an unacknowledged write within a session results in an error?

If this is client-side error (thrown within the driver), then that could be a valuable test. If the server just doesn't support unacknowledged writes with explicit sessions, then I don't think we need to test that behavior on our side.

Comment (prestonvasquez) [reply mongo/integration/causal_consistency_test.go:197]: The test should fail if the `0`-number  write concern being sent over the wire is acknowledged, which will [request[s] no acknowledgment of the write operation](https://docs.mongodb.com/manual/reference/write-concern/#mongodb-writeconcern-writeconcern.-number-).

The error itself is thrown by the `*mongo.Collection.InsertOne` method, specifically whenever the wire message flag is `MoreToCome`.  I can see _[how](https://docs.mongodb.com/manual/reference/mongodb-wire-protocol/#flag-bits)_ we determine this message from [the code base](https://github.com/mongodb/mongo-go-driver/blob/master/x/mongo/driver/wiremessage/wiremessage.go#L336), but have no idea how the relationship between unacknowledged writes and `moreToCome` is 1-1.  The server does not explicitly handle this, and it is [technically recommended behavior](https://github.com/mongodb/specifications/blame/master/source/sessions/driver-sessions.rst#L653).

File: mongo/integration/sessions_test.go [hunk 2]
```diff
@@ -341,6 +342,21 @@ func TestSessions(t *testing.T) {
...
+		sess, err := mt.Client.StartSession()
+		assert.Nil(mt, err, "StartSession error: %v", err)
+		defer sess.EndSession(context.Background())
+
+		err = mongo.WithSession(context.Background(), sess, func(sc mongo.SessionContext) error {
+			_, err := mt.Coll.InsertOne(sc, bson.D{{"x", 1}})
+			return err
+		})
+
+		assert.Equal(mt, err, mongo.ErrUnacknowledgedWrite, "expected error on unacknowledge session write")
```
Comment (matthewdale) [inline mongo/integration/sessions_test.go:358]: Optional: Note that this file imports the `"go.mongodb.org/mongo-driver/internal/testutil/assert"` package, not the `"github.com/stretchr/testify/assert"` package. As a result, the printed failure message doesn't include the actual value comparison failure. Consider adding the actual error value to the error message.

E.g.
```go
assert.Equal(
	mt,
	err,
	mongo.ErrUnacknowledgedWrite,
	"expected ErrUnacknowledgedWrite on unacknowledge session write, got %v",
	err)
```

**Review by matthewdale — APPROVED** [review]
Looks good 👍 with one suggestion.

**Review by benjirewis — APPROVED** [review]
Awesome thanks for the test replacement to maintain coverage 👍

**Review by kevinAlbs — APPROVED** [review]

**Pull request merged**